### PR TITLE
Adds support for collection caching.

### PR DESCRIPTION
**Pull request by djvirgen**
Client updates to support collection caching.

### CODE REVIEW

File: lib/monocle.js [hunk 1]
```diff
@@ -70,13 +76,15 @@ var processBatch = function() {
...
+        body: envelopes
     }).then(function(results) {
         results.forEach(function(result, i) {
             if (result.status >= 200 && result.status < 300) {
                 batched[i].resolve(result.body);
+            } else if (result.status === 304) {
+                batched[i].resolve(batched[i].cached);
             } else {
-                batched[i].reject(result.body);
+                batched[i].reject(resource);
```
Comment (davidkaminsky) [inline lib/monocle.js]: Why reject the resource instead of the response body now?

File: demo/package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "connect": "3.4.0",
     "http-server": "0.8.5",
     "method-override": "2.3.5",
-    "monocle-api": "0.0.5",
+    "monocle-api": "0.2.1",
```
Comment (davidkaminsky) [inline demo/package.json]: Requires 0.2.1 here but 0.1.9 elsewhere?

File: package.json [hunk 3]
```diff
@@ -26,8 +26,9 @@
   "dependencies": {
     "bluebird": "2.10.1",
     "clone": "1.0.2",
-    "monocle-api": "0.1.8",
-    "request": "2.37.0"
+    "monocle-api": "0.1.9",
```
Comment (davidkaminsky) [inline package.json]: 0.1.9 here but 0.2.1 elsewhere?

File: test/test_runner.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ GLOBAL.LIB_DIR = path.join(process.cwd(), 'lib');
 GLOBAL.Promise = require('bluebird');
 
 // Set up in-place instrumentation for code coverage
-require('blanket')({ pattern: LIB_DIR });
+// require('blanket')({ pattern: LIB_DIR });
```
Comment (davidkaminsky) [inline test/test_runner.js]: Should this be commented out?

File: lib/monocle.js (hunk 1 above)
Comment (djvirgen) [inline lib/monocle.js]: Oops! good catch

File: demo/package.json (hunk 2 above)
Comment (djvirgen) [inline demo/package.json]: `0.2.1` is the correct version, both are now updated.

**Comment by marianeaj** [comment]
:dancers:

**Pull request merged**